### PR TITLE
first pass at a typescript declaration file

### PR DESCRIPTION
**Pull request by rokob**
Addresses #302

### CODE REVIEW

**Comment by brianr** [comment]
Nice!

fyi @TimMensch that this is in progress

**Comment by TimMensch** [comment]
Thanks for the update and quick response. :)

**Comment by rokob** [comment]
I'm going to merge this so there is something and we can iterate on it further if there are any issues.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+/*~ This is the module template file for class modules.
+ *~ You should rename it to index.d.ts and place it in a folder with the same name as the module.
+ *~ For example, if you were writing a file for "super-greeter", this
+ *~ file should be 'super-greeter/index.d.ts'
+ */
+
+export = Rollbar;
+
+declare class Rollbar {
+    constructor(options: Rollbar.Configuration?);
```
Comment (Agamnentzar) [inline index.d.ts]: should be `constructor(options: Rollbar.Configuration | undefined);`

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    static init(options: Rollbar.Configuration): Rollbar;
+
+    public global(options: Rollbar.Configuration): Rollbar;
+    public configure(options: Rollbar.Configuration): Rollbar;
+    public lastError(): Error?;
```
Comment (Agamnentzar) [inline index.d.ts]: should be `public lastError(): Error | undefined;`

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        logLevel?: Level;
+        reportLevel?: Level;
+        uncaughtErrorLevel?: Level;
+        endpoint?: string;
+        verbose?: boolean;
+        enabled?: boolean;
+        captureUncaught?: boolean;
+        captureUnhandledRejections?: boolean;
+    }
+    export type Callback = (err: Error?, response: object) => void;
```
Comment (Agamnentzar) [inline index.d.ts]: should be `export type Callback = (err: Error | undefined, response: object) => void;`

**Pull request merged**